api/src/main/java/io/opencensus/trace/SpanBuilder.java

```diff
@@ -223,6 +224,129 @@ public final Scope startScopedSpan() {
     return CurrentSpanUtils.withSpan(startSpan(), true);
   }
 
+  /**
+   * Starts a new span and wraps a {@link Runnable} so that it executes with the {@code Span} as the
+   * current {@code Span}, and ends the {@code Span} after the {@code Runnable} is run. See more
+   * details in {@link Tracer#wrap(Span, Runnable)}.
+   *
+   * <p>To avoid leaking spans or dropping tracing events, the returned {@link Runnable} MUST be run
+   * exactly once.
+   *
+   * <p>Any error will end up as a {@link Status#UNKNOWN}.
+   *
+   * <pre><code>
+   * Runnable newRunnable = tracer.spanBuilder("MyRunnableSpan").startSpanAndWrap(myRunnable);
+   * </code></pre>
+   *
+   * <p>It is equivalent with the following code:
+   *
+   * <pre><code>
+   * Span span = tracer.spanBuilder("MyRunnableSpan").startSpan();
+   * Runnable newRunnable = tracer.wrap(span, new Runnable() {
+   *    {@literal @}Override
+   *     public void run() {
+   *       try {
+   *         myRunnable.run();
+   *       } finally {
+   *         span.end();
+   *       }
+   *     }
+   * });
+   * </code></pre>
+   *
+   * @param runnable the {@code Runnable} to run in the {@code Span}.
+   * @since 0.11.0
+   */
+  public final Runnable startSpanAndWrap(final Runnable runnable) {
+    final Span span = startSpan();
+    return CurrentSpanUtils.wrap(span, runnable, true);
+  }
+
+  /**
+   * Starts a new span and wraps a {@link Callable} so that it executes with the {@code Span} as the
+   * current {@code Span}, and ends the {@code Span} after the {@code Callable} is run. See more
+   * details in {@link Tracer#wrap(Span, Callable)}.
+   *
+   * <p>To avoid leaking spans or dropping tracing events, the returned {@link Callable} MUST be
+   * called exactly once.
+   *
+   * <p>Any error will end up as a {@link Status#UNKNOWN}.
+   *
+   * <pre><code>
+   * {@code Callable<MyResult>} newCallable =
+   *     tracer.spanBuilder("MyCallableSpan").startSpanAndWrap(myCallable);
+   * </code></pre>
+   *
+   * <p>It is equivalent with the following code:
+   *
+   * <pre><code>
+   * Span span = tracer.spanBuilder("MyCallableSpan").startSpan();
+   * {@code Callable<MyResult>} newCallable = tracer.wrap(span, {@code new Callable<MyResult>()} {
+   *    {@literal @}Override
+   *     public MyResult call() throws Exception {
+   *       try {
+   *         return myCallable.call();
+   *       } finally {
+   *         span.end();
+   *       }
+   *     }
+   * });
+   * </code></pre>
+   *
+   * @param callable the {@code Callable} to run in the {@code Span}.
+   * @since 0.11.0
+   */
+  public final <V> Callable<V> startSpanAndWrap(final Callable<V> callable) throws Exception {
```

Comment:
this startSpanAndWrap stuff feels a bit odd to me, unless you are trying to do a lambda trace or something. Particularly wrapping a callable (ex for use in an executorService) usually would defer the start. Hard to see why we optimize for this as there isn't a lot of use cases to look at.

Reply:
The other 2 APIs are very useful, to move this PR forward I will remove this API and we can consider to add it back later if people ask for.